requirements.txt

```diff
@@ -1,5 +1,40 @@
-flask
+attrs==19.3.0
+certifi==2020.4.5.2
+cffi==1.14.0
+chardet==3.0.4
+click==7.1.2
+cryptography==2.9.2
+Flask==1.1.2
+Flask-JWT-Extended==3.24.1
 # pin to last hash of master, to remove requirement of freestanding JSON config file
 git+https://github.com/puiterwijk/flask-oidc.git@7f16e27#egg=flask-oidc
-gunicorn
-requests
+flask-pytest==0.0.5
+gunicorn==20.0.4
+httplib2==0.18.1
+idna==2.9
+importlib-metadata==1.6.1
+itsdangerous==1.1.0
+Jinja2==2.11.2
+jmespath==0.10.0
+MarkupSafe==1.1.1
+more-itertools==8.4.0
+nodeenv==1.4.0
+oauth2client==4.1.3
+packaging==20.4
+pluggy==0.13.1
+py==1.8.2
+pyasn1==0.4.8
+pyasn1-modules==0.2.8
+pycparser==2.20
+PyJWT==1.7.1
+pyparsing==2.4.7
+pytest==5.4.3
```

Comment:
without the top-level dependency list (eg flask, gunicorn, pytest), it will be impossible to distinguish top-level dependencies and their dependencies (which may change)

Reply:
i've just been importing the need, and running freeze.  what's a better approach?

Reply:
perhaps i shouldn't just freeze, but ONLY include the top level ones I've added?

Reply:
>i've just been importing the need
I think you're roughly doing this?:
1. `pip install flask pytest requests` (install top-level deps in venv)
2. `pip freeze` > requirements.txt

What you're doing is working great, but I'm not sure where you're keeping the first list?

We've [used `install_requires` in `setup.cfg`](https://github.com/uwcirg/truenth-portal/blob/develop/setup.cfg#L41), but just another requirements file (for top-level deps only) would be good enough.

We just need to *occasionally* regenerated requirements.txt from scratch from top-level deps.

([`pip-tools compile` scripts some of that process](https://github.com/jazzband/pip-tools), but still requires top-level deps to be separate)

Reply:
unfortunately, i'm *not* keeping the first list.  just tell me your preference, `setup.cfg`?  which project is doing the pattern you like?

Reply:
I went ahead and did this in a followup PR :)
https://github.com/uwcirg/cosri-patientsearch/pull/7